### ISSUE TITLE
undefined method `arity' for nil:NilClass on an have_db_index with unique option

### ISSUE THREAD

**Issue by fidalgo** [thoughtbot/shoulda-matchers#901]
When I have an unique option in a index I have the following message:

```
     NoMethodError:
       undefined method `arity' for nil:NilClass
```

I have provided an  sample application where you can find this behaviour:
https://github.com/fidalgo/shoulda_bug_unique_index

just run `rake spec` and you'll get:

> Failures:
> 
>   1) User db indexes 
>      Failure/Error: it { should have_db_index(:email).with_options(unique: true) }
>      
>      NoMethodError:
>        undefined method `arity' for nil:NilClass
>      # ./spec/models/user_spec.rb:6:in `block (4 levels) in <top (required)>'
> 
>   2) User db indexes 
>      Failure/Error: it { should have_db_index(:reset_password_token).with_options(unique: true) }
>      
>      NoMethodError:
>        undefined method `arity' for nil:NilClass
>      # ./spec/models/user_spec.rb:7:in `block (4 levels) in <top (required)>'
> 
> Finished in 0.01478 seconds (files took 1.15 seconds to load)
> 13 examples, 2 failures
> 
> Failed examples:
> 
> rspec ./spec/models/user_spec.rb:6 # User db indexes 
> rspec ./spec/models/user_spec.rb:7 # User db indexes 
>

**Comment by fidalgo** [comment]
Sorry my mistake to not keep a close eye on [documentation](http://matchers.shoulda.io/docs/v3.1.1/Shoulda/Matchers/ActiveRecord.html#have_db_index-instance_method):

```
  context 'indexes' do
    it { should have_db_index(:name).unique }
  end
``

**Comment by Tashows** [comment]
I faced that issue too. Apparently I hadn't installed `fastimage`. Setting `image_magick` didn't work even after I updated it to latest version. Even after installing `fastimage` I had to set it as the analyzer. Seemed like it wasn't selected by default as mentioned in the docs, however I didn't dig into the matter any deeper. So `Shrine.plugin :store_dimensions, analyzer: :fastimage` made everything work.
